nixos/modules/virtualisation/amazon-init.nix

```diff
@@ -75,6 +75,11 @@ in {
       after = [ "multi-user.target" ];
       requires = [ "network-online.target" ];
 
+      path = [
+        "/run/wrappers"
+        "/run/current-system/sw"
```

Comment:
Note:

```
      path = mkOption {
        default = [];
        type = with types; listOf (oneOf [ package str ]);
        description = ''
          Packages added to the service's {env}`PATH`
          environment variable.  Both the {file}`bin`
          and {file}`sbin` subdirectories of each
          package are added.
        '';
      };
```

suggests `/bin` should not be necessary

Reply:
For _packages_.  But also for strings?

Reply:
Yep!

Here is from a machine with this patch as an AMI:

```
# systemctl cat amazon-init  | cat
# /etc/systemd/system/amazon-init.service
[Unit]
After=multi-user.target
Description=Reconfigure the system from EC2 userdata on startup
Requires=network-online.target
X-StopOnRemoval=false

[Service]
Environment="LOCALE_ARCHIVE=/nix/store/mlya4575shc0a73ssr0j2maf168wyzar-glibc-locales-2.39-52/lib/locale/locale-archive"
Environment="PATH=/run/wrappers/bin:/run/current-system/sw/bin:/nix/store/k71apxkm38m3g34k01sb6zhysi0y7gph-coreutils-9.5/bin:/nix/store/asgsrh9ijxmypmxi0dy3f5bwy84bmjp7-findutils-4.9.0/bin:/nix/store/xpq4yhadyhazkcsggmqd7rsgvxb3kjy4-gnugrep-3.11/bin:/nix/store/5ppp0xw11navs501xv805jwxgn5yrymf-gnused-4.9/bin:/nix/store/109w24xsb42fali6vh051vd2ghiphj4s-systemd-255.9/bin:/run/wrappers/sbin:/run/current-system/sw/sbin:/nix/store/k71apxkm38m3g34k01sb6zhysi0y7gph-coreutils-9.5/sbin:/nix/store/asgsrh9ijxmypmxi0dy3f5bwy84bmjp7-findutils-4.9.0/sbin:/nix/store/xpq4yhadyhazkcsggmqd7rsgvxb3kjy4-gnugrep-3.11/sbin:/nix/store/5ppp0xw11navs501xv805jwxgn5yrymf-gnused-4.9/sbin:/nix/store/109w24xsb42fali6vh051vd2ghiphj4s-systemd-255.9/sbin"
Environment="TZDIR=/nix/store/jq5vhjvin5mgilzy7dr9hfcwg5jxxg1s-tzdata-2024a/share/zoneinfo"
X-RestartIfChanged=false
ExecStart=/nix/store/cr26b6y5aqjzbw31707y5azjffiyy306-unit-script-amazon-init-start/bin/amazon-init-start
RemainAfterExit=true
Type=oneshot

[Install]
WantedBy=multi-user.target
```

Specifically:

**Environment="PATH=/run/wrappers/bin:/run/current-system/sw/bin:**